### PR TITLE
PERF: use `ModData` instead of `RsMod` in `VisibilityFilter`

### PR DESCRIPTION
**Pull request by vlad20012**
changelog: Slightly speed up name resolution

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/resolve2/CrateDefMap.kt [hunk 1]
```diff
@@ -293,6 +293,7 @@ class ModData(
     val name: String get() = path.name
     val isDeeplyEnabledByCfg: Boolean get() = isDeeplyEnabledByCfgOuter && isEnabledByCfgInner
     val parents: Sequence<ModData> get() = generateSequence(this) { it.parent }
+    val isHangingModData: Boolean get() = context != null
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/core/resolve2/CrateDefMap.kt:296]:
```suggestion
    val isHanging: Boolean get() = context != null
```

File: src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt [hunk 2]
```diff
@@ -488,28 +488,32 @@ private fun <T> Map<String, T>.entriesWithNames(names: Set<String>?): Map<String
...
+                val visible = modOpenedInEditor.modData.parents.contains(inMod)
+                return if (visible) VisibilityStatus.Visible else VisibilityStatus.Invisible
             }
         }
     }
     // cfg-disabled items should resolve only from cfg-disabled modules
     if (this === Visibility.CfgDisabled) {
-        return filter@{ context ->
-            val file = context.containingFile as? RsFile ?: return@filter VisibilityStatus.Visible
+        return fun (context: RsElement, _: Lazy<RsModInfo?>?): VisibilityStatus {
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt:508]:
```suggestion
        return fun(context: RsElement, _: Lazy<RsModInfo?>?): VisibilityStatus {
```

File: src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt [hunk 3]
```diff
@@ -488,28 +488,32 @@ private fun <T> Map<String, T>.entriesWithNames(names: Set<String>?): Map<String
...
     if (this is Visibility.Restricted) {
-        val inMod = inMod.toRsMod(info)
-        if (inMod.isNotEmpty()) {
-            return { context ->
-                val modOpenedInEditor = context.containingModOrSelf
-                val visible = inMod.any(modOpenedInEditor.superMods::contains)
-                if (visible) VisibilityStatus.Visible else VisibilityStatus.Invisible
+        val inMod = inMod
+        if (!inMod.isHangingModData) {
+            return fun (context: RsElement, lazyModInfo: Lazy<RsModInfo?>?): VisibilityStatus {
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt:495]:
```suggestion
            return fun(context: RsElement, lazyModInfo: Lazy<RsModInfo?>?): VisibilityStatus {
```

**Review by dima74 — APPROVED** [review]

**Comment by vlad20012** [comment]
bors r=dima74

**Pull request merged**